addons/binding/org.openhab.binding.yamahareceiver/src/main/java/org/openhab/binding/yamahareceiver/internal/protocol/HttpXMLSendReceive.java

```diff
@@ -0,0 +1,258 @@
+    private Logger logger = LoggerFactory.getLogger(YamahaZoneThingHandler.class);
+
+    public final static String XML_GET = "<?xml version=\"1.0\" encoding=\"utf-8\"?><YAMAHA_AV cmd=\"GET\">";
+    public final static String XML_PUT = "<?xml version=\"1.0\" encoding=\"utf-8\"?><YAMAHA_AV cmd=\"PUT\">";
+    public final static String XML_END = "</YAMAHA_AV>";
+
+    /**
+     * Set this flag to write out all send and received traffic
+     */
+    private boolean protocol_sniffer_enabled = false;
+    private FileOutputStream debug_out_stream;
+
+    public enum ReadWriteEnum {
+        GET,
+        PUT
+    }
+
+    public HttpXMLSendReceive(String host) {
+        this.host = host;
+        // Enable sniffing of the communication to the AVR if the logger level is set to trace
+        // when the addon is being loaded.
+        setProtocolSnifferEnable(logger.isTraceEnabled());
+    }
+
+    public void setHost(String host) {
+        this.host = host;
+    }
+
+    public void setProtocolSnifferEnable(boolean enable) {
+        if (enable) {
+            File path_without_file = new File(ConfigConstants.getUserDataFolder());
+            path_without_file.mkdirs();
+            File file = new File(path_without_file, "yamaha_trace.log");
+            if (file.exists()) {
+                file.delete();
+            }
+            logger.warn("Protocol sniffing for Yamaha Receiver Addon is enabled. Performance may suffer! Writing to "
+                    + file.getAbsolutePath());
+            try {
+                debug_out_stream = new FileOutputStream(file);
+            } catch (FileNotFoundException e) {
+                debug_out_stream = null;
+                logger.error(e.getLocalizedMessage());
+            }
+        } else if (protocol_sniffer_enabled) {
+            // Close stream if protocol sniffing will be disabled
+            try {
+                debug_out_stream.close();
+            } catch (Exception e) {
+            }
+            debug_out_stream = null;
+        }
+        this.protocol_sniffer_enabled = enable;
+    }
+
+    static Node getNode(Node root, String nodePath) {
+        String[] nodePathArr = nodePath.split("/");
+        return getNode(root, nodePathArr, 0);
+    }
+
+    private static Node getNode(Node parent, String[] nodePath, int offset) {
+        if (parent == null) {
+            return null;
+        }
+        if (offset < nodePath.length - 1) {
+            return getNode(((Element) parent).getElementsByTagName(nodePath[offset]).item(0), nodePath, offset + 1);
+        } else {
+            return ((Element) parent).getElementsByTagName(nodePath[offset]).item(0);
+        }
+    }
+
+    public static boolean childNodeExists(Node parent, String childNodeName) {
+        return ((Element) parent).getElementsByTagName(childNodeName).item(0) != null;
+    }
+
+    /**
+     * Post the given xml message and return the response as an xml document node.
```

Comment:
To me this method does not look like it's posting anything it seems just to convert an xml string to a Document. If you add docblocks please make them useful and correct